irods/manager/data_object_manager.py

```diff
@@ -121,23 +121,29 @@ def call___del__if_exists(super_):
         next_finalizer_in_MRO()
 
 
-class ManagedBufferedRandom(io.BufferedRandom):
+def managed_class(cls):
 
-    def __init__(self, *a, **kwd):
-        # Help ensure proper teardown sequence by storing a reference to the session,
-        # if provided via keyword '_session'.
-        self._iRODS_session = kwd.pop("_session", None)
-        super(ManagedBufferedRandom, self).__init__(*a, **kwd)
-        import irods.session
+    class managed(cls):
 
-        with irods.session._fds_lock:
-            irods.session._fds[self] = None
+        def __init__(self, *a, **kwd):
+            # Help ensure proper teardown sequence by storing a reference to the session,
+            # if provided via keyword '_session'.
+            self._iRODS_session = kwd.pop("_session", None)
+            super(managed, self).__init__(*a, **kwd)
+            import irods.session
 
-    def __del__(self):
-        if not self.closed:
-            self.close()
-        call___del__if_exists(super(ManagedBufferedRandom, self))
+            with irods.session._fds_lock:
+                irods.session._fds[self] = None
 
+        def __del__(self):
+            if not self.closed:
+                self.close()
+            call___del__if_exists(super(managed, self))
+
+    return managed
+
+m_BufferedRandom = managed_class(io.BufferedRandom)
+m_iRODSDataObjectFileRaw = managed_class(iRODSDataObjectFileRaw)
```

Comment:
Are these private to the implementation? i.e. Should users of the library ever reach for this? If not, consider adding a leading underscore to signal that to the reader.